packages/daf-core/src/graphql/graphql-core.ts

```diff
@@ -113,52 +116,190 @@ function transformFindInput(input: FindInput): FindManyOptions {
       if (item.not === true) {
         where[item.column] = Not(where[item.column])
       }
+      return where
+    }
+  }
+}
+
+function decorateQB(
+  qb: SelectQueryBuilder<any>,
+  tablename: string,
+  input: FindInput,
+): SelectQueryBuilder<any> {
+  if (input?.skip) qb = qb.skip(input.skip)
+  if (input?.take) qb = qb.take(input.take)
+
+  if (input?.order) {
+    for (const item of input.order) {
+      qb = qb.orderBy(`"${tablename}"."${item.column}"`, item.direction)
     }
-    result['where'] = where
   }
-  return result
+  return qb
 }
 
 export interface FindArgs {
   input?: FindInput
 }
 
 const messages = async (_: any, args: FindArgs, ctx: Context) => {
-  return (await ctx.agent.dbConnection).getRepository(Message).find(transformFindInput(args.input))
+  const where = createWhereObject(args.input)
+  let qb = (await ctx.agent.dbConnection)
+    .getRepository(Message)
+    .createQueryBuilder('message')
+    .where(where)
+  qb = decorateQB(qb, 'message', args.input)
+  if (ctx.agent.authentication) {
```

Comment:
Since this authentication is impacting only GraphQL resolvers, I would suggest not adding this logic to the `Agent` and keeping it only in GraphQL `Context`. Like this:



``` typescript
export interface Context {
  agent: Agent,
  authenticatedDid: string,
}

// ...

  if (ctx.authenticatedDid) {
    qb = qb.andWhere(
      new Brackets(qb => {
        qb.where('message.to = :ident', { ident: authenticatedDid })
        .orWhere('message.from = :ident', { ident: authenticatedDid })
      }),
    )
  }
```

This way we leave it to application developers to decide how they authenticate requests.

Reply:
ok, needs to be Promise<string> because context function can't be async